dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -84,6 +87,12 @@ String getAggFuncName(
         return count_second_stage;
     }
 
+    // sum function in mpp has two stage and we need to distinguish them with function name.
+    // "sum" represents the first stage.
+    // "finalSumStage" represents the sencond stage.
```

Comment:
from `count_second_stage`, it looks like the second stage is a the partial stage.

Reply:
agg does not always have two stages, such as complete mode agg.

Reply:
Do we need to do the same for partial agg?
If the type deduction of decimal between two aggs is inconsistent, will cast be introduced between the two aggs?

Reply:
such as `final agg <-- decimal cast in beforeAggStreams <-- exchange_receiver <-- partial agg`

Reply:
> such as `final agg <-- decimal cast in beforeAggStreams <-- exchange_receiver <-- partial agg`



> Do we need to do the same for partial agg? If the type deduction of decimal between two aggs is inconsistent, will cast be introduced between the two aggs?

So far, I haven't seen there is any inconsistent between partial agg and final agg. So, it's needless to do this.

Reply:
> agg does not always have two stages, such as complete mode agg.

Yes, but initialization of partial agg and complete agg is the same. We only need to initialize the FinalMode sum in a different way.